external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -1427,17 +1434,45 @@ impl SymbolicatorRunner {
                                 }
                                 continue;
                             }
-                            pkgs_to_analyze
-                                .entry(root_dir.unwrap())
-                                .or_insert_with(BTreeSet::new)
-                                .insert(starting_path.clone());
+                            // we want to collect either all modified Move files (as BTreeSet<Some(PathBuf)> or,
+                            // if one of the paths for a package is not a Move file, have aonly one BTreeSet
+                            // None entry. This will be later passed to symbolicator as Some(Vec<PathBuf>) or
+                            // None, to indicate if incremental symbolication is possible or not.
+                            let root_dir = root_dir_opt.unwrap();
+                            if let Some(mut modified_files) = pkgs_to_analyze.remove(&root_dir) {
+                                // we already seen this package
+                                if modified_files.len() != 1 || modified_files.first().is_some() {
+                                    // keep adding modfied Move files if any
+                                    if is_move_file(starting_path) {
+                                        // add move file to existing set
+                                        modified_files.insert(Some(starting_path.clone()));
+                                        pkgs_to_analyze.insert(root_dir, modified_files);
+                                    } else {
+                                        // reset to a single None entry
+                                        pkgs_to_analyze.insert(root_dir,  BTreeSet::from_iter(vec![None]));
+                                    }
+                                } // else we already seen this package with a non-Move-file path so nothing to do
+                            } else {
+                                // first time we see this package
```

Comment:
It seems like this logic could be combined with the above logic if we used `entry`

Reply:
Logic changed so no longer needed